testing/trino-testing/src/main/java/io/trino/testing/AbstractTestFailureRecovery.java

```diff
@@ -105,30 +128,54 @@ public void testAggregation()
     @Test(invocationCount = INVOCATION_COUNT)
     public void testJoin()
     {
-        testSelect("SELECT * FROM orders o, customer c WHERE o.custkey = c.custkey AND c.nationKey = 1");
+        @Language("SQL") String selectQuery = "SELECT * FROM partitioned_lineitem JOIN supplier ON partitioned_lineitem.suppkey = supplier.suppkey " +
+                "AND supplier.name = 'Supplier#000000001'";
+        testSelect(selectQuery, Optional.of(enableDynamicFiltering(false)), queryId -> {});
+
+        testSelect(
+                selectQuery,
+                Optional.of(enableDynamicFiltering(true)),
+                queryId -> {
+                    DynamicFiltersStats dynamicFiltersStats = getDynamicFilteringStats(queryId);
+                    assertThat(dynamicFiltersStats.getLazyDynamicFilters()).isEqualTo(1);
+                    DynamicFilterDomainStats domainStats = getOnlyElement(dynamicFiltersStats.getDynamicFilterDomainStats());
+                    assertThat(domainStats.getSimplifiedDomain())
+                            .isEqualTo(singleValue(BIGINT, 1L).toString(getSession().toConnectorSession()));
+                    OperatorStats probeStats = searchScanFilterAndProjectOperatorStats(queryId, getQualifiedTableName(PARTITIONED_LINEITEM));
+                    assertThat(probeStats.getInputPositions()).isIn(615L, 1230L);
```

Comment:
why not exact value? Add comment

Reply:
added comment
The reason is that currently stats from all attempts are combined. I'm asserting on multiple of 615L as well in case the probe scan was completed twice

Reply:
Is there a TODO to fix that?

Reply:
@arhimondr had mentioned there is a plan for updating reporting of stats for retries but I'm not sure whether that involves giving operator stats per attempt

Reply:
It is possible to make it fail after previous TS finishes (e.g make test deterministic?)

Reply:
`testSelect` internally injects failure in 4 different ways, i'm not sure whether we can deterministically fail or not fail probe scan in all 4 failure scenarios.
Alternatively, I can dilute the assertion to be `probeStats.getInputPositions() < LINEITEM_COUNT`. Let me know if that's better.